### ISSUE TITLE
Some questions in reproducing the result of NVAN network

### ISSUE THREAD

**Issue by yuange250** [jackie840129/STE-NVAN#6]
Dear Jackie:
     Thanks for your generous in sharing your fantastic codes.
     I have some questions in reproducing the results of NVAN network. I just follows the training instruction:
     `python3 evaluate.py --test_txt $TEST_TXT  --test_info  $TEST_INFO   --query_info $QUERY_INFO \
                    --batch_size 64 --model_type 'resnet50_NL' --num_workers 8  --S 8 --latent_dim 2048 \
                    --temporal Done  --non_layers  0 2 3 0 --load_ckpt $LOAD_CKPT \`
     And for other superparameters, I just follows the default settings in parser.py.
     I run the codes on 4 * 1080Ti, then I only get the result:
            CMC: 0.8672, mAP : 0.7946
    Which is far from the result reported in the paper.
    Is there something wrong on my operations? I spot that in the validation of NVAN, you just select 8 frames from each video sequence, the other frames are just abandon, is that the main reason on the results decreasing？
    Hoping for your generous answer~
    Best regards!

**Comment by jackie840129** [comment]
Hi,
So you trained the model from scratch and test it with my script? Or you just test it with the trained model provided by me?
I'm really sure that it can reach the performance if you train it from scratch. Can you provide the training script you used?  Thanks!

**Comment by yuange250** [comment]
Thanks for your quick responce, I train the model derectly on the train_NL.py, and the parameters is as follows:
`

def parse_args():
    parser = argparse.ArgumentParser(description='Train Video-based Re-ID',formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument('--train_txt',help='txt for train dataset', default="/home/chenzy/NVAN/STE-NVAN-master/MARS_database/train_path.txt")
    parser.add_argument('--train_info',help='npy for train dataset', default="/home/chenzy/NVAN/STE-NVAN-master/MARS_database/train_info.npy")
    parser.add_argument('--test_txt',help='txt for test dataset', default="/home/chenzy/NVAN/STE-NVAN-master/MARS_database/test_path.txt")
    parser.add_argument('--test_info',help='npy for test dataset', default="/home/chenzy/NVAN/STE-NVAN-master/MARS_database/test_info.npy")
    parser.add_argument('--query_info',help='npy for test dataset', default="/home/chenzy/NVAN/STE-NVAN-master/MARS_database/query_IDX.npy")
    parser.add_argument('--lr',type=float,default=0.0001,help='learning rate')
    parser.add_argument('--lr_step_size',type=int,default=50,help='step size of lr')
    parser.add_argument('--class_per_batch',type=int,default=8)
    parser.add_argument('--track_per_class',type=int,default=4)
    parser.add_argument('--batch_size',type=int,default=64)
    parser.add_argument('--n_epochs',type=int,default=200)
    parser.add_argument('--num_workers',type=int,default=16)
    parser.add_argument('--S',type=int,default=8)
    parser.add_argument('--latent_dim',type=int,default=2048,help='resnet50:2048,densenet121:1024,densenet169:1664')
    parser.add_argument('--load_ckpt',type=str,default=None)
    parser.add_argument('--log_path',type=str,default='loss.txt')
    parser.add_argument('--ckpt',type=str,default="/data/chenzy/models/mars")
    parser.add_argument('--optimizer',type=str,default='adam')
    parser.add_argument('--resume_validation',type=bool,default=False)
    parser.add_argument('--model_type',type=str,default='resnet50_NL')
    parser.add_argument('--stride',type=int,default=1)
    parser.add_argument('--temporal',default='Done')
    parser.add_argument('--frame_id_loss',action='store_true',default=False)
    parser.add_argument('--track_id_loss',action='store_true',default=True)
    parser.add_argument('--non_layers',default=[0, 2, 3, 0], nargs='+')
    parser.add_argument('--stripes',type=int, nargs='+')
    # parser.add_argument(
    args = parser.parse_args()
    return args


`

**Comment by jackie840129** [comment]
Hi @yuange250 ,
The arguments are all correct. I guess that because you train on four GPUs, the batchnorm is performed on the 1/4 data in a batch. It may harm the statistics. (For every batch, I will sample 8 ID, 4 tracks per ID, so you only train 2 ID on one GPU)

You can first test my trained model to ensure that it can achieve 90%, and maybe you can try some combination (like S=6, or try non_layers = '0 1 2 0') to make it on two GPUs. The accuracy can also be near 88~89%.

**Comment by yuange250** [comment]
Thanks!

**Comment by huge123** [comment]
@jackie840129  I want to know what kind of GPU you used to train? It seem that single GPU cannot handle that batchsize in default setting
